crates/amaru/src/ledger/state.rs

```diff
@@ -188,64 +184,71 @@ impl<S: Store<Error = E>, E: std::fmt::Debug> State<S, E> {
 }
 
 /// Process a given block into a series of ledger-state diff (a.k.a events) to apply.
-fn apply_block(span: &Span, block: MintedBlock<'_>) -> VolatileState<()> {
+fn apply_block(parent: &Span, block: MintedBlock<'_>) -> VolatileState<()> {
     let failed_transactions = FailedTransactions::from_block(&block);
+    let transaction_bodies = block.transaction_bodies.to_vec();
+    let total_count = transaction_bodies.len();
+    let failed_count = failed_transactions.inner.len();
```

Comment:
Pardon my Haskell here; I avoided using `.len` to avoid traversing the vector twice, which is why  those numbers were counted during the traversals. But I just double-checked and turns out `Vec` actually holds onto its length, so `len()` is merely just reading a variable. So, all good 👍